addons/hr_contract/models/base_action_rule.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class BaseActionRule(models.Model):
+    """ Add resource and calendar for time-based conditions """
+
+    _inherit = 'base.action.rule'
+
+    trg_date_resource_field_id = fields.Many2one('ir.model.fields', string='Use employee work schedule',
+            help='Use the user\'s working schedule.')
+
+    @api.v8
+    def _check_delay(self, record, record_dt):
```

Comment:
@hmo-odoo if put this type of decorator means its never change in new api ??
its just fixing not migration.